### ISSUE TITLE
ACK doesnt work on Oracle Open Message Queue5.1 (Glassfish 4.1)

### ISSUE THREAD

**Issue by staabm** [stomp-php/stomp-php#35]
I am getting error messages in the simples scenarios when I try to ACK a frame:

```php
$client = new Client('tcp://javasandbox:7672');
$client->setLogin('xy', 'xy');
$client->getConnection()->setReadTimeout(5);
$stomp = new SimpleStomp($client);
$stomp->subscribe('/queue/myqueue', null, 'client-individual');
$frame = $stomp->read();
var_dump($frame);
$stomp->ack($frame);
var_dump("ACK");
$frame = $stomp->read();
```

The error reads
`ACK: [BSS4028]: ACK a message ID\c81124-127.0.1.1(da\c98\cc\c7b\c6b\cc4)-1-1451290154466 that is not found in subscriber`

the actual message id in the queue is `ID:81124-127.0.1.1(da:98:c:7b:6b:c4)-1-1451290154466`, which makes me wonder that the client seem to replace some `:` with `\`.

Forcing the ACK `Frame` into `legacyMode` makes things work.

**Comment by staabm** [comment]
After reading the SPEC of STOMP 1.1[1] and 1.0[2] I found something which might be the problem.

ATM `stomp-php` escapes some characters no matter which version of stomp is used for the connection.
In stomp 1.0 the only char to be escaped is LF and in header-name additionally the `:`.
Escaping of `:` in the header-value was added with stomp 1.1.

Since my queue is talking stomp 1.0 but values are escaped like in 1.1+ I get this problems.

[1] http://stomp.github.io/stomp-specification-1.1.html#Value_Encoding
[2] https://stomp.github.io/stomp-specification-1.0.html#augmented-bnf
